src/core/utils/qfieldcloudutils.cpp

```diff
@@ -58,11 +60,52 @@ bool QFieldCloudUtils::isCloudAction( const QgsMapLayer *layer )
 
   const QString layerAction( layer->customProperty( QStringLiteral( "QFieldSync/cloud_action" ) ).toString().toUpper() );
 
-  if ( layerAction == QStringLiteral( "NO_ACTION" ) || layerAction == QStringLiteral( "REMOVE" ) )
-    return false;
+  if ( isFileBasedLayer( layer ) )
+  {
+    if ( layerAction == QStringLiteral( "REMOVE" ) )
+    {
+      return false;
+    }
+  }
+  else
+  {
+    if ( layerAction == QStringLiteral( "NO_ACTION" ) || layerAction == QStringLiteral( "REMOVE" ) )
+    {
+      return false;
+    }
+  }
+
   return true;
 }
 
+bool QFieldCloudUtils::isFileBasedLayer( const QgsMapLayer *layer )
+{
+  if ( layer->type() == Qgis::LayerType::VectorTile )
+  {
+    QgsDataSourceUri uri;
+    uri.setEncodedUri( layer->source() );
+    return QFile::exists( uri.param( "url" ) );
+  }
+
+  if ( !layer->dataProvider() )
+  {
+    return false;
+  }
+
+  QgsProviderMetadata *providerMetadata = QgsProviderRegistry::instance()->providerMetadata( layer->dataProvider()->name() );
+  QVariantMap metadata = providerMetadata->decodeUri( layer->source() );
+
+  const QString filename = metadata.value( QStringLiteral( "path" ), QString() ).toString();
+  const QString resolvedFilename = QgsProject::instance()->pathResolver().writePath( filename );
+
+  if ( resolvedFilename.startsWith( QStringLiteral( "localized:" ) ) )
```

Comment:
Do we track changes on localized layers?

Reply:
Thanks for raising this. I guess we do not want to track localized layers. I think it would be nice if we mark them as localized somehow, e.g. in the layer tree as we do with locked, tracking and similar right-hand indicatiors. This way the user should be made aware that all edits will not go through QFC.

Reply:
Would it be reasonable to make them read only?

Reply:
Though about this, but making them read-only is too much of a restriction IMO. I can imagine one editing the same gpkg from multiple projects OR using the shared localized path as basemap gpkg that once in a while needs updating on the field.

Reply:
I do not follow sorry. If they are not tracked, how can they be updated from multiple devices?

Reply:
Multiple projects, not devices. One can do this manually via cable.

Reply:
Ah, do we even sync these through qfc yet? That might be it.

Reply:
Not yet, but soon to be.